pywhat/Data/regex.json

```diff
@@ -1309,6 +1309,23 @@
          ]
       }
    },
+   {
+      "Name": "UPS Tracking Number",
+      "Regex": "^(1Z[0-9A-Z]{6}[0-9]{2}[0-9]{8})$",
+      "plural_name": false,
+      "Description": null,
+      "Rarity": 1,
```

Comment:
I would say that rarity should be lowered.

Reply:
Something around 0.3

Reply:
Why 0.3? I'd say higher like 0.5 or 0.6 because:
1. The string has to start with `1Z`
2. It needs 7 chars `0-9A-Z`
3. It has exactly 2 numbers
4. It has 8 numbers

Also, can we make it:

```diff
- ^(1Z[0-9A-Z]{6}[0-9]{2}[0-9]{8})$
- + ^(1Z[0-9A-Z]{6}[0-9]{10})$
```

?

Reply:
I think 0.4 or 0.5. And yes, regex should be changed.

Reply:
The idea of the 2+8 split is because the first 2 digits in this group represent a service indicator code and perhaps it could be captured and handled in the future.

Reply:
Aside: I wonder if the "rarity" could be estimated more reliably through some entropy-based measure 🤔

Reply:
> service indicator code

We have precedence for this called sub-categories. See the Mastercard / Phone Numbers regex. I am not sure it'll work on data in the middle of the regex, we may need to change the code for that :)  

> Aside: I wonder if the "rarity" could be estimated more reliably through some entropy-based measure 🤔

Probably! Currently I am estimating it based on what I see when people post this:
<img width="676" alt="Screenshot 2021-11-04 at 16 09 24" src="https://user-images.githubusercontent.com/10378052/140376319-6687d169-c06c-4a73-9dcd-d632b6b19c8c.png">

And also whether we have any false positives.

Reply:
@P403n1x87 You can use subcategories with regex method for that.